autoload/clap/provider/git_diff_files.vim

```diff
@@ -0,0 +1,23 @@
+" Author: KITAGAWA Yasutaka <kit494way@gmail.com>
+" Description: List the files which is managed by git and have uncommitted changes.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+let s:git_diff_files = {}
+
+function! s:git_diff_files.source() abort
+  if !executable('git')
+    return ['git executable not found']
+  endif
+
+  return map(systemlist('git status -s -uno'), 'split(v:val)[-1]')
```

Comment:
According to the help of `systemlist()`, it will return an empty string on error, then applying a map on it will raise an E712 error.

```suggestion
  let untracked = systemlist('git status -s -uno')
  if untracked ==# ''
     return ['Error occurs on calling `git status -s -uno`, maybe you are not in a git repo.']
  else
    return map(untracked, 'split(v:val)[-1]')
  endif
```

Reply:
Thanks for your suggestion.
I missed error handling.
But `untracked ==# ''` causes an error when `systemlist()` succeeds.
So, based on the way of [commits provider](https://github.com/liuchengxu/vim-clap/blob/6045f807046a1dec301f7573ab2ead5e4fde705a/autoload/clap/provider/commits.vim#L9),  I changed to check `v:shell_error` and use `clap#helper#echo_error()` to report error.

Reply:
Actually it's ok to return the error message directly as they will be shown in the display window. But I'm also fine with just printing the error :(.